amethyst_network/src/connection.rs

```diff
@@ -62,10 +59,7 @@ impl<E: Send + Sync + 'static> NetConnection<E> {
 
 impl<E> PartialEq for NetConnection<E> {
     fn eq(&self, other: &Self) -> bool {
-        self.target_receiver == other.target_receiver
-            && self.state == other.state
-            && self.target_sender == other.target_sender
-            && self.state == other.state
+        self.target_addr == other.target_addr && self.state == other.state
```

Comment:
I want to say that addr should be the only thing here, since there can only be one connection with said address, therefore its unique.